app/validators/source_object_checksum_validator.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+# Ensures that this object's associated source_object has been assigned a fixity_checksum_value.
+class SourceObjectChecksumValidator < ActiveModel::Validator
+  def validate(record)
+    return if record.source_object&.fixity_checksum_value.present?
```

Comment:
```suggestion
    return if (record.source_object&.fixity_checksum_value.present? && record.source_object&.fixity_checksum_algorithm_id.present? )
```

Reply:
Will update in upcoming commit! (plus separate validation messages for missing value vs. algorithm)